db.js

```diff
@@ -149,7 +149,19 @@ var friends = (function () {
 
 var users = (function () {
   return {
-    get: function (userId) {
+    getUserByName: function (userName) {
+      var expr = "'" + userName + "%'";
+      var sql = "SELECT * FROM users WHERE name LIKE " + expr;
+
+      return pool.query(sql)
+        .then(([rows, fields]) => {
+          if (!rows || rows.length === 0) {
+            return null;
+          }
+          return rows;
```

Comment:
Since you're essentially checking if rows is null on 158, then returning null explicitly, you might as well just return rows. ie. `.then(([rows, fields] => { return rows;})`